core/src/services/onedrive/builder.rs

```diff
@@ -0,0 +1,151 @@
+/// Currently, only OneDrive Personal is supported.
+/// For uploading, only files under 4MB are supported via the Simple Upload API (<https://learn.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_put_content?view=odsp-graph-online>).
+///
+/// # Configuration
+///
+/// - `access_token`: set the access_token for Graph API
+/// - `root`: Set the work directory for backend
+///
+/// You can refer to [`OneDriveBuilder`]'s docs for more information
+///
+/// # Example
+///
+/// ## Via Builder
+///
+/// ```no_run
+/// use anyhow::Result;
+/// use opendal::services::Onedrive;
+/// use opendal::Operator;
+///
+/// #[tokio::main]
+/// async fn main() -> Result<()> {
+///     // create backend builder
+///     let mut builder = Onedrive::default();
+///
+///     builder
+///         .access_token("xxx")
+///         .root("/path/to/root");
+///
+///     let op: Operator = Operator::new(builder)?.finish();
+///     Ok(())
+/// }
+/// ```
+#[derive(Default)]
+pub struct OnedriveBuilder {
+    access_token: Option<String>,
+    root: Option<String>,
+    http_client: Option<HttpClient>,
+}
+
+impl Debug for OnedriveBuilder {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut de = f.debug_struct("Builder");
+        de.field("endpoint", &self.access_token);
+        de.finish()
+    }
+}
+
+impl OnedriveBuilder {
+    /// set the bearer access token for OneDrive
+    ///
+    /// default: no access token, which leads to failure
+    pub fn access_token(&mut self, access_token: &str) -> &mut Self {
+        self.access_token = Some(access_token.to_string());
+        self
+    }
+
+    /// Set root path of OneDrive folder.
+    pub fn root(&mut self, root: &str) -> &mut Self {
+        self.root = Some(root.to_string());
+        self
+    }
+
+    /// Specify the http client that used by this service.
+    ///
+    /// # Notes
+    ///
+    /// This API is part of OpenDAL's Raw API. `HttpClient` could be changed
+    /// during minor updates.
+    pub fn http_client(&mut self, http_client: HttpClient) -> &mut Self {
+        self.http_client = Some(http_client);
+        self
+    }
+}
+
+impl Builder for OnedriveBuilder {
+    fn build(&mut self) -> Result<Self::Accessor> {
+        let root = normalize_root(&self.root.take().unwrap_or_default());
+        debug!("backend use root {}", root);
+
+        let client = if let Some(client) = self.http_client.take() {
+            client
+        } else {
+            HttpClient::new().map_err(|err| {
+                err.with_operation("Builder::build")
+                    .with_context("service", Scheme::Onedrive)
+            })?
+        };
+
+        match self.access_token.clone() {
+            Some(access_token) => Ok(OnedriveBackend::new(root, access_token, client)),
+            None => Err(Error::new(ErrorKind::ConfigInvalid, "access_token not set")),
+        }
+    }
+
+    const SCHEME: Scheme = Scheme::Onedrive;
+
+    type Accessor = OnedriveBackend;
```

Comment:
Please keep the order for better reading.

```
const SCHEME
type Accessor

fn from_map
fn build
```

Reply:
Fixed. Thank you!